sdk/core/azure_core_test/src/recording.rs

```diff
@@ -120,10 +138,25 @@ impl Recording {
         self.test_mode
     }
 
-    /// Gets the named variable from the environment or recording.
-    pub fn var(&self, name: impl AsRef<str>) -> Option<String> {
+    /// Gets a required variable from the environment or recording.
+    pub fn var<K>(&self, key: K) -> String
+    where
+        K: AsRef<str>,
+    {
+        let key = key.as_ref();
+        self.var_opt(key)
+            .unwrap_or_else(|| panic!("{key} is not set"))
+    }
+
+    /// Gets an optional variable from the environment or recording.
+    pub fn var_opt<K>(&self, key: K) -> Option<String>
+    where
+        K: AsRef<str>,
+    {
+        // TODO: Should we attempt to preface "AZURE_" or "{SERVICE_DIRECTORY}_" like .NET does?
```

Comment:
FWIW, for C++, we ended up [doing both](https://github.com/Azure/azure-sdk-for-cpp/blob/4cbb49f0364fc47cc91bc7b3e7d1945516a1a80e/sdk/core/azure-core-test/inc/azure/core/test/test_base.hpp#L340).

Reply:
Good to know. Thanks!